custom_components/robovac/vacuum.py

```
@@ -904,6 +1008,456 @@ def _update_cleaning_stats(self) -> None:
+            return
+        if self._attr_room_names:
+            return
+        if self.hass is None:
+            return
+
+        self._cloud_room_lookup_attempted = True
+        try:
+            cloud_rooms = await self.hass.async_add_executor_job(
+                self._fetch_room_names_from_eufy_oauth
+            )
+        except Exception as err:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)